doc/problem/behavior.md

```diff
@@ -52,3 +62,41 @@ For In/Active State:
 - AS Probability: Already in basic time bin features.
 - AS Duration: Already in basic time bin features.
 - AS Rate: Use AS Number over AS Duration.
+
+
+## Data Requirements Description
+The data we have:
+- a dataframe of observations of location for each mice, (x, y, t) with \Delta t small.
+- The above dataframe with a classification of strain number and mouse number at each time t (if available)
+
+The data we require:
+- The above dataframe with AS/ IS properly classified at each point t for each mouse
+- The above dataframe with a classification of drinking, feeding and locomotion for each mouse at each time t
+- The above dataframe with a classification of consumption size and moving distance with each event at each time t
+
+## Methodology/ Approach Description
+We wish to create a single function that should be able to return all of the above metrics as a list:
+- Key inputs are:
+    - mouse/ strain as string
+    - starting time
+    - ending time
+    - a dictionary containing the rectangular vertices marking the area to restrict the movement to i.e. x_lower, x_upper, y_lower, y_upper.
+    - [_Drinking | Feeding | Locomotion_] state specification
+- The main output is a list containing the key metrics stated in `Statement of statistical problems` section
+
+- Key idea is that if we have the most granular dataframe in `Data Requirements Description` then the python code is really just a SQL (in `pandas` form) filtering/ grouping query to generate the required output metrics (from flowchart) in the form of a list
+
+## Testing Framework Outline
+
+## Additional Remarks
+- It is not clear exactly how the existing metrics are to be calculated in the form a single query
```

Comment:
This sentence doesn't totally make sense. Perhaps it should say "in order to form a single query" ?